EasyCommands/CommandParsers/ParameterParsers.cs

```diff
@@ -464,19 +464,19 @@ partial class Program {
                     : new Token[] { new Token(element, true, characters.Length > 1) })  // Keep the entire item
                 .ToArray();
 
-        String[] ParseSeparateTokens(String command) {
-            var newCommand = command;
-            foreach (var token in separateTokensFirstPass) newCommand = newCommand.Replace(token, " " + token + " ");
-            newCommand = string.Join(" ", newCommand.Split(new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries).SelectMany(token => {
+        String[] ParseSeparateTokens(String command) =>
+            AddSpaceArroundTokens(command, separateTokensFirstPass)
+            .Split(new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries)
+            .SelectMany(token => {
                 Primitive ignored;
                 if (separateTokensFirstPass.Contains(token) || ParsePrimitive(token, out ignored)) return new[] { token };
-                foreach (var s in separateTokensSecondPass) token = token.Replace(s, " " + s + " ");
-                return token.Split(new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
-            }));
-
-            List<char> commandArray = newCommand.ToCharArray().ToList();
+                return AddSpaceArroundTokens(token, separateTokensSecondPass).Split(new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
Nice!  nit: can we use the ternary operator to remove the two "return" statements?

return (separateTokensFirstPass.Contains(token) || ParsePrimitive(token, out ignored)) ? new[] { token} : AddSpaceAroundTokens(....

?